test/modules/offload-s3/offload_test.go

```diff
@@ -182,6 +191,15 @@ func Test_Upload_DownloadS3Journey(t *testing.T) {
 			}, 5*time.Second, time.Second, fmt.Sprintf("tenant was never %s", models.TenantActivityStatusFROZEN))
 		})
 
+		t.Run("verify tenant data version is 1 after freezing", func(t *testing.T) {
```

Comment:
also better to lock the behavior of "multiple clients try to freeze the tenant concurrently" and still version is incremented once?

Reply:
good idea, thanks! i added checks to the `Test_ConcurrentFreezeUnfreeze` func below